Data/ByteString/Builder/RealFloat/F2S.hs

```diff
@@ -70,11 +76,99 @@ mulShift32 m factor shift =
 
 -- | Index into the 64-bit word lookup table float_pow5_inv_split
 get_float_pow5_inv_split :: Int -> Word64
+#if !PURE_HASKELL
 get_float_pow5_inv_split = getWord64At float_pow5_inv_split
+#else
+-- > putStr $ case64 (finv float_pow5_inv_bitcount) [0..float_max_inv_split]
+get_float_pow5_inv_split i = case i of
```

Comment:
I wonder if these tables are too large to be inlined, even though they aren't exported. It could be worthwhile to just spot check the Core since these tables are likely to optimize with case-of-known-constructors.

Reply:
These tables are used too deep in the call hierarchy for case-of-known-constructors to apply.

I've looked at Core for `formatDouble scientific 1.0` with `-O2` and 9.8.1:
- first it's stuck on the worker for `formatDouble` (`$wformatDouble`) that isn't inlined
- if we inline it, it's then stuck on `d2s` that isn't inlined
- if we inline it, we get a lot of bloat (~300 lines) and the compile-time reduction is stuck on:

```haskell
  = case {__ffi_static_ccall_safe base:stg_doubleToWord64zh "stg_doubleToWord64zh" :: Double#
                                                                             -> Word64#}_ii3E
           1.0##
    of ds1_ii3H
    { __DEFAULT ->
    GHC.Word.W64# ds1_ii3H
    }  
```

which is a shame and should be fixed. I've opened https://gitlab.haskell.org/ghc/ghc/-/issues/24331 to track this.

In the meantime I won't try to inline these.